### DISCUSSION TITLE
Crazyflie logging variable missing cf_lib_Python

### DISCUSSION THREAD

**Discussion by Wenyu-Wu** [bitcraze/discussions#368] [Q&A]
We are trying to log several variable values from Crazyflie through cf_lib_Python. We keep getting errors indicating variables are not in TOC. Upon checking with cfclient, variables like posCtl.Xi seem to be missing. Is there a way to edit the firmware for them to be available? Variables we are currently looking for are 'posCtl.Xi', 'posCtl.VYi' (and more). 
![error_msg](https://user-images.githubusercontent.com/59187434/200070695-6b4aa1d1-450d-4b5c-8ac6-3a0aded6009a.png)

**Comment by krichardsson** [comment]
Those logs were added in the beginning of October, are you perhaps running a firmware that is older?
Building your firmware from master should solve the problem.

**Reply by Wenyu-Wu** [reply]
Thanks for replying. We have tried to clone the repo "crazyflie-firmware", made it a zip, and flashed it to the drone with cfclient. The issue still seems to persist. Are there issues with the procedure we are currently following? Thanks!

**Reply by krichardsson** [reply]
Even though it is possible to create a zip and flash using the client, the easiest way is to use `make cload` from a terminal instead, see [the flashing instructions in the documentation](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/#flashing) for more information.

**Reply by Wenyu-Wu** [reply]
We have just tried this. We are still not seeing the proper TOC. Are there other options we can look into? Thanks!

**Reply by krichardsson** [reply]
The `posCtl.VYi` log is [defined here](https://github.com/bitcraze/crazyflie-firmware/blob/5be4c083aa6700ee4e79286eb6d755e22520b5be/src/modules/src/position_controller_pid.c#L424). If you have it in your source code, build and flash, it should appear in the log TOC. 
Make sure you are building the correct source files and that you flash the correct binary.
